File: front-veterinaria/src/app/Services/usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Usuario, UsuarioCustom, UsuarioMapper } from '../Models/usuario';
import { API_URL } from '../Environments/routes';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(API_URL.usuarios.find_all);
  }

  findAllCustom(): Observable<UsuarioCustom[]> {
    return this.http.get<UsuarioCustom[]>(API_URL.usuarios.find_all_custom);
  }

  findAllMapper(): Observable<UsuarioMapper[]> {
    return this.http.get<UsuarioMapper[]>(API_URL.usuarios.find_all_mapper);
  }

  findById(id: number): Observable<Usuario> {
    return this.http.get<Usuario>(API_URL.usuarios.find_by_id(id));
  }

  insert(usuario: Usuario): Observable<any> {
    return this.http.post<any>(API_URL.usuarios.insert, usuario);
  }

  update(id: number, usuario: Usuario): Observable<any> {
    return this.http.put<any>(API_URL.usuarios.update(id), usuario);
  }

  delete(id: number): Observable<any> {
    return this.http.delete<any>(API_URL.usuarios.delete(id));
  }
}

File: back-veterinaria/src/main/java/idat/proyecto/veterinaria/service/UsuarioService.java
package idat.proyecto.veterinaria.service;

import org.springframework.http.ResponseEntity;

import idat.proyecto.veterinaria.entity.Usuario;

public interface UsuarioService {
	
	public abstract ResponseEntity<?> insert(Usuario usuario);
	public abstract ResponseEntity<?> update(Integer id, Usuario usuario);
	public abstract ResponseEntity<?> delete(Integer id);
	public abstract ResponseEntity<?> findById(Integer id);
	public abstract ResponseEntity<?> isExistUsername(String username);
	public abstract ResponseEntity<?> isExistCorreo(String correo);
	public abstract ResponseEntity<?> findAll();
	public abstract ResponseEntity<?> findAllCustom();
	public abstract ResponseEntity<?> findAllMapper();
}
